src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -472,9 +487,18 @@ void ObjectSynchronizer::enter(Handle obj, BasicLock* lock, JavaThread* current)
     if (LockingMode == LM_LIGHTWEIGHT) {
       // Fast-locking does not use the 'lock' argument.
       LockStack& lock_stack = current->lock_stack();
-      if (lock_stack.can_push()) {
-        markWord mark = obj()->mark_acquire();
-        while (mark.is_neutral()) {
+      if (lock_stack.is_full()) {
+        // The emitted code always goes into the runtime in case the lock stack
+        // is full. We unconditionally make room on the lock stack by inflating
+        // the least recently locked object on the lock stack.
+        ObjectMonitor* monitor = inflate(current, lock_stack.bottom(), inflate_cause_vm_internal);
+        assert(monitor->owner() == current, "must be owner=" PTR_FORMAT " current=" PTR_FORMAT " mark=" PTR_FORMAT,
+               p2i(monitor->owner()), p2i(current), monitor->object()->mark_acquire().value());
+        assert(!lock_stack.is_full(), "must have made room here");
+      }
+
+      markWord mark = obj()->mark_acquire();
+      while (mark.is_neutral()) {
           // Retry until a lock state change has been observed.  cas_set_mark() may collide with non lock bits modifications.
           // Try to swing into 'fast-locked' state.
           assert(!lock_stack.contains(obj()), "thread must not already hold the lock");
```

Comment:
It looks like the indent from L502 -> L515 is too much by two spaces.
This could be a GitHub view glitch...

Reply:
Done. (Occurred when merging).